### DISCUSSION TITLE
How to change LICENSE for a repository?

### DISCUSSION THREAD

**Discussion by penyuan** [community/community#23555] [New to GitHub]
<p>I know it is easy to add a license to a GitHub repository for the first time and have done so for all of mine. However, if a repository already has a license, what’s a good way to change it?</p>
<p>Do I simply remove the existing LICENSE file, change any and all SPDX identifiers in files, and GitHub will automatically give me the option to add a LICENSE again?</p>
<p>Also, what’s a good way to propagate the license change to all branches in a repository?</p>
<p>Thanks!</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/penyuan">@penyuan</a>, thank you for being here! You are correct you can remove the file and then check your community profile, and you’ll see the option to add a license once again.</p>
<p>As far as the branches, we only look at the license on the default branch, and apply that to the whole repository.</p>
<p>I hope this helps!</p>

**Reply by nenamoree9** [reply]
it doesnt work

**Reply by Axlmj2006** [reply]
> Hi [@penyuan](/penyuan), thank you for being here! You are correct you can remove the file and then check your community profile, and you’ll see the option to add a license once again.
> 
> As far as the branches, we only look at the license on the default branch, and apply that to the whole repository.
> 
> I hope this helps!

**Comment by penyuan** [comment]
<aside class="quote group-github-employees" data-username="AndreaGriffiths11" data-post="2" data-topic="126844">
<div class="title">
<div class="quote-controls"></div>
 AndreaGriffiths11:</div>
<blockquote>
<p>You are correct you can remove the file and then check your community profile, and you’ll see the option to add a license once again.</p>
<p>As far as the branches, we only look at the license on the default branch, and apply that to the whole repository.</p>
</blockquote>
</aside>
<p>Awesome, that’s all great to know. Thank you so much for your quick and concise reply! :smile:</p>

**Comment by kunalrshah** [comment]
<p>I never selected a license. But, I now wish to add it or select it from the list of choices. How do I get that option back? Adding a LICENSE.md file didn’t work for me.</p>

**Reply by amoldhamale1105** [reply]
Just click on add file option and create a new file with the name "LICENSE"
Once you do that GitHub will prompt you with various licensing options

**Comment by penyuan** [comment]
<p><a class="mention" href="/kunalrshah">@kunalrshah</a>: Try removing <code>LICENSE.md</code> or <code>LICENSE</code> from your repository. In my case, GitHub offered me an option to choose a license soon after that.</p>

**Reply by SashaFloress** [reply]
you can open the license file, then select `edit this file` from drop down menu of edit, on the upper right side you will see `Choose a license Template` a new page will pops up with all available licenses on a left side panel to choose from then `review and submit`.
This will change license in main branch, you then have to adopt new license above all pragma line.

**Comment by akashdip2001** [comment]
If you want to change the license of an existing GitHub repository, you can follow these steps:

1. **Remove the Existing License:**
   - Delete the existing `LICENSE` file in your repository. This file usually contains the text of the license and may be named differently depending on the license type.

2. **Update SPDX Identifiers:**
   - Go through the source code and documentation in your repository and update any references to the old license's SPDX identifier.

3. **Commit Changes:**
   - Commit your changes to the repository.

4. **Add New License:**
   - GitHub should automatically detect the absence of a license file and provide an option to add a new license.
   - Navigate to your repository on GitHub.
   - Click on "Add file" and then choose "Create new file."
   - Name the file `LICENSE` and select the license template you want to use.
   - Commit the new `LICENSE` file.

5. **Propagation to Other Branches:**
   - If you have multiple branches, the license change should be applied automatically when you merge or rebase branches. However, you can ensure consistency by following these steps:
     - If you are working with branches, make the license change in one branch, then merge or rebase other branches onto the branch with the license change.
     - If you are working with forks, make the change in the upstream repository, and contributors can update their forks accordingly.

Remember to communicate the license change to contributors and users of your repository. Changing a license is a significant action, and it's essential to ensure that everyone is aware of and agrees with the change.

## Profile
![Profile Image](https://thumbnails.odycdn.com/optimize/s:200:0/quality:95/plain/https://thumbs.odycdn.com/bf58cf4d7ec52c62f02c52e16f150a5b.webp)
**AKASHDIP MAHAPATRA**
"everything is physics bro"  
E=mc²

|[Subscribe on YouTube](https://www.youtube.com/channel/UCxvmp634YDc41xCWOdvWqoQ?sub_confirmation=1) |
|----  |

**Comment by tellmehey** [comment]
thanks 4 the article

**Comment by Napster7-0** [comment]
Hey!
to add a license to an existing project, you can go to "insights" near "settings", and select "Community Standards", there your project is compared to an recommended community standard; Naturally a standard project has a license. If Your own doesn't have it, choose "Add a license", choose a license and commit changes.

For sure, the previous answers are also a way to got it!

**Reply by yairEO** [reply]
this tab isn't always there, so one can try clicking the far-right 3-dots menu button which can be found on the same level as the top tabs are 

![image](https://github.com/user-attachments/assets/68f7285c-5d27-46e0-ac54-5e5ba6cf3a0f)

**Reply by Napster7-0** [reply]
You're right.

**Comment by indigo-jay** [comment]
I created a new repo, added files, etc. But the repo shows it's a `GPL-2.0 license` even though there is no LICENSE file. How do tell GitHub that this repo has no license?

**Reply by indigo-jay** [reply]
Found my own answer - one file in the repo has a @license attribution at the top, specifying that it's `GPL 2`.

**Comment by kcamacho1** [comment]
From the repository look on the right hand column. There are small words "Readme, LICENSE, Star, etc..." click License > then click the edit icon > then click "Select License".

**Comment by native-api-work** [comment]
As of 2025, you need to delete the existing `LICENSE` file, then add it again via Github UI. After you type an appropriate name for the file in the new file editor form, there will appear a "Choose a license template" button in the editor's header.

See [Adding a license to a repository - GitHub Docs](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository) for details.
